File: ea/src/evolution_engine/fitness/FitnessResult.java
package evolution_engine.fitness;

import evolution_engine.evolution.Individual;

import java.util.Map;

/**
 * Created By Itay Azaria
 * Date: 9/16/2014
 */
public interface FitnessResult<I extends Individual> {
    public double getFitness(I individual);
}

File: src/main/java/evolution_engine/fitness/FitnessResult.scala
package evolution_engine.fitness

import evolution_engine.evolution.Individual

/**
 * Created By Itay Azaria
 * Date: 9/16/2014
 */
trait FitnessResult[I <: Individual] {
  def getFitness(individual: I): Double

  def getMap: Map[I, Double]
}